src/main/resources/application.conf

```diff
@@ -38,110 +46,224 @@ etc-client {
         "enode://fba5a07e283d517a2680bcfc7aeb498ac2d246d756556a2ebd5edeb39496491c47a6d27e27f82833b7d7d12defc8de994de04bb58beb72472649f9a323006820@41.135.121.6:30303"
       ]
 
+      # Time interval for scanning bootstrap nodes
+      # TODO: explain better
       bootstrap-nodes-scan-interval = 2 minutes
     }
 
     peer {
+      # Retry delay for failed attempt at connecting to a peer
       connect-retry-delay = 20 seconds
+
+      # Maximum number of attempts to connect to a peer
+      # TODO: what happens after that? blacklisting?
       connect-max-retries = 30
+
+      # TODO: what's this for? why delay a PoisonPill
       disconnect-poison-pill-timeout = 5 seconds
+
+      # Timeout for 'Hello' message from peer (during handshake)
       wait-for-hello-timeout = 3 seconds
+
+      # Timeout for 'Status' message from peer (during handshake)
       wait-for-status-timeout = 30 seconds
+
+      # Timeout for block header message that determines the current Ethereum fork (during handshake)
       wait-for-chain-check-timeout = 15 seconds
 
+      # Maximum block headers in a single message
       max-blocks-headers-per-message = 200
+
+      # Maximum block bodies in a single message
       max-blocks-bodies-per-message = 200
+
+      # Maximum transactions receipts in a single message
       max-receipts-per-message = 200
+
+      # Maximum MPT components in a single message
       max-mpt-components-per-message = 400
 
+      # Maximum connected peers
       max-peers = 10
+
+      # Ethereum network identifier:
+      # 1 - mainnet, 2 - morden
+      # TODO: other ids?
       network-id = 1
     }
 
     rpc {
+      # Whether to enable JSON-RPC HTTP endpoint
       enabled = true
+
+      # Listening address of JSON-RPC HTTP endpoint
       interface = "127.0.0.1"
+
+      # Listening port of JSON-RPC HTTP endpoint
       port = 8546
+
+      # Enabled JSON-RPC APIs over the HTTP endpoint
+      # Available choices are: eth, web3, net, personal
       apis = "eth,web3,net"
     }
   }
 
   txPool {
+    # Maximum number of pending transaction kept in the pool
     tx-pool-size = 1000
+
+    # Timeout for querying the transaction pool
+    # TODO: do we have 2 timeouts for this? This and mining.pooling-services-timeout ?
     pending-tx-manager-query-timeout = 5.seconds
```

Comment:
as discussed, this is the timeout for Akka ask, very internal thing. I think it doesn't need documenting.